hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HealthTransformations.scala

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.monster.dap
+
+import org.broadinstitute.monster.dogaging.jadeschema.table.HlesHealthCondition
+
+object HealthTransformations {
+
+  /** Parse all health-condition-related fields out of a raw RedCap record. */
+  def mapHealthConditions(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
+    // condition type is congenital vs infectious vs tox vs trauma vs etc
+    health_conditions.filter { case (condition, _) => rawRecord.getBoolean(s"hs_${condition}_yn") }.flatMap {
+      // infectious disease
+      case ("dx_infectious", conditionCategorical) =>
+        // process infectious disease data
+        infectious_diseases.filter {
+          case (infection, _) => rawRecord.getBoolean(s"hs_dx_${infection}")
+        }.map {
+          case (infection, infectionCategorical) =>
+            HlesHealthCondition(
+              dogId = rawRecord.getRequired("study_id").toLong,
+              hsConditionType = conditionCategorical.toLong,
+              hsCondition = infectionCategorical.toLong,
+              // should only be populated if hs_dx_infect_other is the current infectionCategorical (value is 40)
+              hsConditionOtherDescription = if (infectionCategorical == 40) {
```

Comment:
We shouldn't need to have this "magic" dependency between constants. We can tag each of the field names with a field for other / not other, or we could leave the "other" case out of the list and have a special one-off call for it.